tests/basis_test.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+from basis.basis import BasisFunction, ChebyshevFirstKind
+
+@pytest.fixture
+def expected_extrema_2():
```

Comment:
How were these generated? You should note the source, and if possible, write them using exact math rather than decimals.

Reply:
They were generated from the old code files that Mohammadreza and I made earlier in the summer. I can represent everything up except some of the extrema for exactness 4 in mathematical terms.

I have found that ones like 1/sqrt(2) are not floating point precise with the answer the code gives, but they are very close.

Reply:
Gotcha, I think it is probably sufficient to go to exactness of 3 so you can write them all down “exactly”. Then, your unit tests should use floating point comparison with a tolerance.